lib/services/elasticsearch.js

```diff
@@ -222,41 +222,37 @@ class ElasticSearch extends Service {
       scroll
     };
 
-    debug('Search: %o', esRequest);
+    debug('Search: %j', esRequest);
 
-    return this._client.search(esRequest)
-      .then(({ body }) => {
-        if (body._scroll_id) {
-          const
-            // ms(scroll) may return undefined if in microseconds or in nanoseconds
-            ttl = esRequest.scroll && ms(esRequest.scroll)
-              || ms(this._config.defaults.scrollTTL),
-            key = scrollCachePrefix + this._kuzzle.constructor.hash(
-              body._scroll_id
-            );
-
-          return this._kuzzle.cacheEngine.internal.psetex(key, ttl, 0)
-            .then(() => this._formatSearchResult(body));
-        }
+    let body;
+    try {
+      ({ body } = await this._client.search(esRequest)); // NOSONAR
```

Comment:
It's annoying to have to add this comment because Sonarqube does not like this syntax.

Reply:
Did you try:
```js
       { body } = (await this._client.search(esRequest)); // NOSONAR 
```
Maybe this one is allowed by Sonar :slightly_smiling_face:

Reply:
This syntax is not valid javascript unfortunately :(

Reply:
I already notified Florient to ask if he could update our sonarqube servers, it's stuck to a very old version.
In the meantime... yeah, more and more modern JS gets rejected by sonar